Algorithm.Framework/Portfolio/AccumulativeInsightPortfolioConstructionModel.cs

```diff
@@ -0,0 +1,243 @@
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Alphas;
+using QuantConnect.Data.UniverseSelection;
+
+namespace QuantConnect.Algorithm.Framework.Portfolio
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IPortfolioConstructionModel"/> that allocates percent of account
+    /// to each insight, defaulting to 3%.
+    /// For insights of direction <see cref="InsightDirection.Up"/>, long targets are returned and
+    /// for insights of direction <see cref="InsightDirection.Down"/>, short targets are returned.
+    /// No rebalancing shall be done, as a new insight or the age of the insight shall determine whether to exit
+    /// the positions.
+    /// Rules:
+    ///    1. On Up insight, increase position size by percent
+    ///    2. On Down insight, decrease position size by percent
+    ///    3. On Flat insight, move by percent towards 0
+    ///    4. On expired insight, perform a Flat insight'''
+    /// </summary>
+    public class AccumulativeInsightPortfolioConstructionModel : PortfolioConstructionModel
+    {
+        private List<Symbol> _removedSymbols = new List<Symbol>();
+        private readonly InsightCollection _insightCollection = new InsightCollection();
+        private DateTime? _nextExpiryTime;
+        private readonly double _percent;
+        private readonly Dictionary<Insight, int> usedInsight = new Dictionary<Insight, int>();
+        private readonly Dictionary<Insight, int> expiredList = new Dictionary<Insight, int>();
+        private readonly Dictionary<Symbol, double> positionSizes = new Dictionary<Symbol, double>();
+                
+        /// <summary>
+        /// Initialize a new instance of <see cref="AccumulativeInsightPortfolioConstructionModel"/>
+        /// </summary>
+        /// <param name="percent">The amount of portfolio to allocate to a single insight</param>
+        public AccumulativeInsightPortfolioConstructionModel(double percent = 0.03)
+        {
+            _percent = Math.Abs(percent);
+        }
+
+       
+        /// <summary>
+        /// Method that will determine if the portfolio construction model should create a
+        /// target for this insight
+        /// </summary>
+        /// <param name="insight">The insight to create a target for</param>
+        /// <returns>True if the portfolio should create a target for the insight</returns>
+        public virtual bool ShouldCreateTargetForInsight(Insight insight)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Will determine the target percent for each insight
+        /// </summary>
+        /// <param name="activeInsights">The active insights to generate a target for</param>
+        /// <returns>A target percent for each insight</returns>
+        public Dictionary<Insight, double> DetermineTargetPercent(ICollection<Insight> activeInsights)
```

Comment:
Should make this parameter `IEnumerable<Insight>`. In general, you want to select the most general and least restrictive interface possible. In this case, the only action performed on `activeInsights` is enumeration, which is best communicated via this `IEnumerable<T>` interface.

Reply:
also -- does this method need to be public? it looks like an internal implementation detail.